react/features/invite/components/dial-in-info-page/DialInInfoPage.web.js

```diff
@@ -0,0 +1,217 @@
+     */
+    state = {
+        conferenceID: null,
+        error: '',
+        loading: true,
+        numbers: null,
+        numbersEnabled: null
+    };
+
+    /**
+     * Initializes a new {@code DialInInfoPage} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        // Bind event handlers so they are only bound once for every instance.
+        this._onGetNumbersSuccess = this._onGetNumbersSuccess.bind(this);
+        this._onGetConferenceIDSuccess
+            = this._onGetConferenceIDSuccess.bind(this);
+        this._setErrorMessage = this._setErrorMessage.bind(this);
+    }
+
+    /**
+     * Implements {@link Component#componentDidMount()}. Invoked immediately
+     * after this component is mounted.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidMount() {
+        const getNumbers = this._getNumbers()
+            .then(this._onGetNumbersSuccess)
+            .catch(this._setErrorMessage);
+
+        const getID = this._getConferenceID()
+            .then(this._onGetConferenceIDSuccess)
+            .catch(this._setErrorMessage);
+
+        Promise.all([ getNumbers, getID ])
+            .then(() => {
+                this.setState({ loading: false });
+            });
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        let contents;
+
+        const { conferenceID, error, loading, numbersEnabled } = this.state;
+
+        if (loading) {
+            contents = '';
+        } else if (numbersEnabled === false) {
+            contents = this.props.t('invite.disabled');
+        } else if (error) {
+            contents = error;
+        } else {
+            contents = [
+                conferenceID
+                    ? <ConferenceID
+                        conferenceID = { conferenceID }
+                        key = 'conferenceID' />
+                    : null,
+                <NumbersTable
+                    key = 'numbers'
+                    numbers = { this.state.numbers } />
+            ];
+        }
+
+        return (
+            <div className = 'dial-in-page'>
+                { contents }
+            </div>
+        );
+    }
+
+    /**
+     * Creates an AJAX request for the conference ID.
+     *
+     * @private
+     * @returns {Promise}
+     */
+    _getConferenceID() {
+        const { room } = this.props;
+        const { dialInConfCodeUrl, hosts } = config;
+        const mucURL = hosts && hosts.muc;
+
+        if (!dialInConfCodeUrl || !mucURL || !room) {
+            return Promise.resolve();
+        }
+
+        const conferenceIDURL
+            = `${dialInConfCodeUrl}?conference=${room}@${mucURL}`;
+
+        return fetch(conferenceIDURL)
```

Comment:
If the idea of using fetch is okay, I'd like to bring it into the main app and replace all jquery ajax calls with it. I couldn't re-use existing actions for getting the numbers and conference ids because:
1) the current implementation assumes a redux store exists and it looked annoying to bring it into the bundle
2) separating out just the ajax calls somehow might have required bringing in jquery into this separate bundle which wouldn't need jquery for anything else.

I'm open to ideas though on how I could repurpose existing logic to fetch the numbers and conference id.

Reply:
FWIW, we use fetch on mobile (to download the config), so a big +1 from me for using it for ajax calls.